research/context/preselection/preselect_wikipedia_comments.py

```diff
@@ -0,0 +1,176 @@
+tf.app.flags.DEFINE_integer("randomly_selected_limit", 100, "Number of comments selected at random, not based on toxicity.")
+tf.app.flags.DEFINE_string("comment_type", "ADDITION", "The type of comment (one of 'ADDITION', 'DELETION', 'CREATION', 'MODIFICATION').")
+tf.app.flags.DEFINE_integer("min_txt_len", 1, "Minimum length for comments of the sample.")
+tf.app.flags.DEFINE_integer("max_txt_len", 1000, "Maximum length for comments of the sample.")
+tf.app.flags.DEFINE_string("bq_table_name", None, "The name of the Google BigQuery table.")
+tf.app.flags.DEFINE_string("bq_project_name", None, "The name of the Google BigQuery project.")
+tf.app.flags.DEFINE_string("output_csv_path", None, "Path where the output sample CSV should be written. Either that or JSON should not be empty.")
+tf.app.flags.DEFINE_string("output_json_path", None, "Path where the output sample JSON should be written. Either that or CSV should not be empty.")
+
+def connect_to_bq(bq_project_name):
+    """
+    Connect to Google BigQuery
+    :param bq_project_name: project name
+    :return:
+    """
+    client = bigquery.Client(project=bq_project_name)
+    return client
+
+def query(low_tox, high_tox, limit, max_txt_len, min_txt_len, type, bq_table_name):
+    """
+    Google Big Query sampling data of specific types
+    :param low_tox: comments should be higher than that
+    :param high_tox: comments should be lower or equal than that
+    :param limit: fetch some
+    :param type: kind of comments (wikipedia specific)
+    :param max_txt_len: max length of comment
+    :param bq_table_name: the table name in BigQuery
+    :return: dataframe with results
+    """
+    assert low_tox>=0 and high_tox<=1
+    query = """
+    SELECT *, rand() as r 
+        FROM `{bq_table_name}`
+        WHERE length(cleaned_content)>{min_txt_len} AND length(cleaned_content)<{max_txt_len} AND replyTo_id!="null" AND type="{typ}" AND toxicity>={low} AND toxicity<{high}
+        ORDER BY r
+        LIMIT {lim}
+    """.format(bq_table_name=bq_table_name, low=low_tox, high=high_tox, lim=limit, typ=type, max_txt_len=max_txt_len, min_txt_len=min_txt_len)
+    query_job = client.query(query)
+    results = query_job.result()
+    return results.to_dataframe()
+
+
+def sample_based_on_toxicity(limit, max_txt_len, min_txt_len, type, bq_table_name, toxic_zones_num, randomly_selected_limit):
+    """
+    Google Big Query sampling data of specific types
+    :param limit: fetch some
+    :param type: kind of comments (wikipedia specific)
+    :param max_txt_len: max length of comment
+    :param bq_table_name: the table name in BigQuery
+    :param toxic_zones_num: number of toxic zones to be created from 0 to 1
+    :return: dataframe with results
+    """
+    query = """"""
+    if randomly_selected_limit>0:
+        query = """
```

Comment:
Add comment to say that `replyTo_id!="null"` is to make sure you don't get comments that are just section title creation. And note that it could also have been done by typ only being "ADDITION"